src/type_checker/checker.rs

```diff
@@ -641,6 +638,9 @@ impl<B: Backend> TypeChecker<B> {
         Ok(None)
     }
 
+    // todo: collect generic parameters and update FnInfo.generics
+    // - HashSet<Ident, ValueBound>
+    // - ValueBound: Enum { Bound(u32), Unbound }
```

Comment:
why?

Reply:
this todo is outdated. 
the extraction is already done in the `FnSig`